src/app/modules/payments/stripe.controller.ts

```diff
@@ -388,3 +423,98 @@ export const handleConnectOauthCallback = [
   }),
   _handleConnectOauthCallback,
 ] as ControllerHandler[]
+
+export const getPaymentInfo: ControllerHandler<
+  { paymentId: string },
+  GetPaymentInfoDto | ErrorDto
+> = async (req, res) => {
+  const { paymentId } = req.params
+  logger.info({
+    message: 'getPaymentInfo endpoint called',
+    meta: {
+      action: 'getPaymentInfo',
+      paymentId,
+    },
+  })
+
+  return PaymentService.findPaymentById(paymentId)
+    .andThen((payment) => {
+      return PendingSubmissionModel.findPendingSubmissionById(
+        payment.pendingSubmissionId,
+      )
+        .andThen((submission) => FormService.retrieveFormById(submission.form))
+        .andThen((form) => {
+          // Payment forms are encrypted
+          if (!isFormEncryptMode(form)) {
+            logger.error({
+              message:
+                'Requested for payment information in possibly non-payment form. Expected payment forms to be encrypted forms',
+              meta: {
+                action: 'getPaymentInfo',
+                paymentId,
+                formResponseMode: form.responseMode,
+              },
+            })
+            return errAsync(
+              new ResponseModeError(
+                FormResponseMode.Encrypt,
+                form.responseMode,
+              ),
+            )
+          }
```

Comment:
nit: would it be better to use `checkFormIsEncryptMode` instead, so that we can do `.andThen(checkFormIsEncryptMode)`?

Reply:
Cleaned up in 57585489